src/3box.js

```diff
@@ -57,91 +56,61 @@ class ThreeBox {
-
     const keystore = new OrbitdbKeyAdapter(this._muportDID)
     console.time('new OrbitDB')
     this._orbitdb = new OrbitDB(this._ipfs, opts.orbitPath, { keystore })
     console.timeEnd('new OrbitDB')
     globalIPFS = this._ipfs
     globalOrbitDB = this._orbitdb
 
+    this._rootStore = await this._orbitdb.feed(rootStoreName)
+    const rootStoreAddress = this._rootStore.address.toString()
+
     this.public = new PublicStore(this._orbitdb, didFingerprint + '.public', this._linkProfile.bind(this))
     this.private = new PrivateStore(this._muportDID, this._orbitdb, didFingerprint + '.private')
 
-    if (rootStoreAddress) {
-      console.time('open rootStore')
-      this._rootStore = await this._orbitdb.open(rootStoreAddress)
-      console.timeEnd('open rootStore')
-      // this._rootStore.events.on('replicate', console.log)
-      const readyPromise = new Promise((resolve, reject) => {
-        this._rootStore.events.on('ready', resolve)
-      })
-      console.time('rootStore.load')
-      this._rootStore.load()
-      await readyPromise
-      console.timeEnd('rootStore.load')
-      // console.log('p2', (await this._ipfs.swarm.peers())[0].addr.toString())
-      // console.log('p2 id', (await this._ipfs.id()).id)
-      // console.log(this._rootStore.iterator({ limit: -1 }).collect().length)
-      // console.log(await this._ipfs.pubsub.peers('/orbitdb/QmRxUAGk62v7NjUkzvcqwYkBqF3zHb8tfhfW6T3MateGje/b932fe7ab.root'))
-      console.time('replicate rootStore')
-      const entries = this._rootStore.iterator({ limit: -1 }).collect()
-      const rootStoreEmpty = !entries.length
-      const dbSyncPromise = (entry) => {
-        const odbAddress = entry.payload.value.odbAddress
-        const name = odbAddress.split('.')[1]
-        if (name === 'public') return this.public._sync(odbAddress)
-        if (name === 'private') return this.private._sync(odbAddress)
-      }
-      let storePromises = []
-      console.time('sync stores')
-      if (rootStoreEmpty) {
-        await new Promise((resolve, reject) => {
-          this._rootStore.events.on('replicate.progress', (address, hash, entry, progress, have) => {
-            storePromises.push(dbSyncPromise(entry))
-            if (progress === have) {
-              resolve()
-              console.timeEnd('replicate rootStore')
-            }
-          })
-        })
-      } else {
-        entries.forEach(entry => { storePromises.push(dbSyncPromise(entry)) })
+    console.time('init stores')
+    const [pubOdbAddress, privOdbAddress] = await Promise.all([this.public._init(), this.private._init()])
+    console.timeEnd('init stores')
+
+    this._createRootStore(rootStoreAddress, privOdbAddress, pubOdbAddress, pinningNode)
+
+    return {rootStoreAddress, privOdbAddress, pubOdbAddress}
+  }
+
+  async _sync (privOdbAddress, pubOdbAddress) {
+    console.time('sync stores')
+    await Promise.all([this.public._sync(pubOdbAddress), this.private._sync(privOdbAddress)])
```

Comment:
We don't really need the odbAddresses here. the `_sync()` method should assume that `_init()` has been called and can therefore use `this._db` directly.